src/components/usePopup.ts

```diff
@@ -28,14 +93,20 @@ export const usePopup = (): PopupProps => {
     };
 
     const addSelectedCountry = (country: CountryName) => {
-        const newSet = new Set(selectedCountries);
-        newSet.add(country);
-        setSelectedCountries(newSet);
+        const newMap = new Map(selectedCountries);
+        if (newMap.has(country)) {
+            return
+        }
+        newMap.set(country, 0);
+        setSelectedCountries(newMap);
     }
     const removeSelectedCountry = (country: CountryName) => {
-        const newSet = new Set(selectedCountries);
-        newSet.delete(country);
-        setSelectedCountries(newSet);
+        const newMap = new Map(selectedCountries);
+        if (!newMap.has(country)) {
+            return
+        }
+        newMap.delete(country);
```

Comment:
One of the best practices in react is to not make mutable changes, can you think of why?

Reply:
Based on an article encouraging [avoiding mutation altogether](https://blog.sapegin.me/all/avoid-mutation/), and another one where some [mutation is okay](https://dev.to/nas5w/mutation-isnt-always-bad-in-javascript-5e94),

Mutation of objects in JS can be bad because 
- React uses object references to detect changes
  - This may cause React to not detect changes when object props are mutated instead of using a new object
- When mutation occurs from different components, this can lead to unexpected behaviour, making it hard to debug 


1. I'm not quite sure whether this counts as a mutation, since the object we are trying to mutate within this function is selectedCountries, which we are replacing with a completely new map, which is a variable local to this function
2. If this still counts as a mutation, should I be using `const newMap = new Map(selectedCountries).delete(country);` instead? 

What are your thoughts?

Reply:
Yeah, so mutability leads to stuff happening that we dont know about for example division by 0 or even changing types. These are called side effects. This can lead to more bugs in the code base and really annoying debugging scenarios. In the case of this function it might be ok and we can leave it :D